assets/js/blocks/products/all-products/edit.js

```diff
@@ -283,7 +284,7 @@ class Editor extends Component {
 		const blockTitle = this.getTitle();
 		const blockIcon = this.getIcon();
 
-		if ( ! getSetting( 'hasProducts', true ) ) {
+		if ( blocksConfig.productCount === 0 ) {
```

Comment:
If `blocksConfig.productCount` isn't defined this condition won't trigger.

Reply:
Understood. I don't think it matters in this context because we're defining the settings for this block in the block type class.